### PR TITLE
Regen improvements v2

### PR DESCRIPTION
**Pull request by Tom-evnut**
Param:Speed has now been cleaned up in all files, making sure only the use ABS when required.
Created Param to inverter motor rotation - only really applicable to Outlander, but also implemented on Leaf.

Implemented, if pedal is steady no regen allowed but still taper the PotNom on speed base.

Clean up work in GS450h
-Checksum
-Create template for Prius, IS300h and GS450h specifics within functions

clean up work in Leaf
-Propper 15bit decoding of speed
-Added reversemotor

RearOutlander
-Added reversemotor

Param
-Added reversemotor param
      Created Param to inverter motor rotation - only really applicable to Outlander, but also implemented on Leaf.
-Added regenendrpm
     rpm under which regen is forced to 0, to remove jerkyness
-can3speed
    added 100kbps

### CODE REVIEW

**Review by damienmaguire — APPROVED** [review]
Thanks Tom

File: src/RearOutlanderinverter.cpp [hunk 1]
```diff
@@ -23,89 +23,93 @@
...
-
-
-   Param::SetInt(Param::torque,final_torque_request);//post processed final torque value sent to inv to web interface
+    final_torque_request = ((torquePercent * 2000) / 100.0f ); //!! Moved into parameter *-1 reverses torque direction
+
+    if(Param::GetInt(Param::reversemotor) == 1)
+    {
+        final_torque_request *= -1;//reverse torque request to flip motor rotation
+    }
+    final_torque_request += 10000;
```
Comment (Landsi) [inline src/RearOutlanderinverter.cpp:71]: I don't quite get that line. It's not there for the leaf inverter.

Why potentially making negative torque only to add 10000 to it? 🤔

Comment (Tom-evnut) [reply src/RearOutlanderinverter.cpp:71]: For creating the right value to writing to CAN for the inverter. It has a 10000 offset,

Comment (Landsi) [reply src/RearOutlanderinverter.cpp:71]: thx, i now of course see the +10000 above in the original line.

i guess that's why code reviews on github mobile are not a good idea 🤦‍♂️

**Pull request merged**